src/main/java/teammates/common/util/StringHelper.java

```diff
@@ -359,6 +359,177 @@ public static String csvToHtmlTable(String str) {
                              result.toString());
     }
 
+    /**
+     * Combines all the strings inside the passed list using the specified delimiter. The result will be
+     * another list with a single entry, which is the combined string, or an empty list if the passed
+     * argument is an empty list.
+     * @param list of string
+     * @param delimiter the connectors in between the strings
+     * @return list containing the combined string or an empty list
+     */
+    private static List<String> combineListEntries(List<String> list,
```

Comment:
list is a very generic variable name. maybe use stringList?
This function is very similar to the "join" function used in other languages/libraries. maybe naming it "join" would be better.